app/main.py

```
@@ -1 +1,35 @@
-# write your code here
+from __future__ import annotations
+from types import TracebackType
+from typing import Optional, Type
+import os
+
+
+class Cleaner:
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    def __enter__(self) -> Cleaner:
+        return self
+
+    def __exit__(self,
+                 exc_type: Optional[Type[BaseException]],
+                 exc_val: Optional[BaseException],
+                 exc_tb: Optional[TracebackType]
+                 ) -> None:
+        os.remove(self.name)
+
+
+def move_file(command: str) -> None:
+    parts = command.split()
+    if len(parts) == 3:
+        action, source_file, destination = parts
+        if os.path.exists(source_file):
```

The function checks os.path.exists(source_file) and otherwise silently does nothing when the source is missing. The task description/example implies that missing source should lead to standard file errors (e.g., FileNotFoundError). Do not silently return; let open() raise or raise a clear error yourself. See line 26 and the description example .